tests/test_screenshot.py

```diff
@@ -15,6 +17,20 @@ def test_screenshot(browser):
     assert testdir.tmpdir.join('test_browser_screenshot_normal', 'test_screenshot-browser.png').isfile()
 
 
+@patch('_pytest.config.Config.warn')
+def test_browser_screenshot_error(testdir, mocked_browser, request):
+    """Test warning with error during taking screenshots on test failure."""
+    testdir.inline_runsource("""
+        def test_screenshot(browser):
+            # Create a file here, so makedirs in make_screenshot_on_failure will fail.
+            open('isafile', 'w').close()
+            assert False
+
+        def test_warn_called(request):
+            assert request.config.warn.call_count == 1
```

Comment:
I've tried that first, but it did not work - the warning gets done in the inner test.

I've committed it in 008728f, and added an sentence to the doc.